service/nodes.go

```diff
@@ -36,13 +34,6 @@ func newNodeID(parts ...string) nodeID {
 	return nodeID(h.Sum64())
 }
 
-type componentNode interface {
-	component.Component
-	graph.Node
-}
-
-var _ componentNode = &receiverNode{}
-
 // A receiver instance can be shared by multiple pipelines of the same type.
 // Therefore, nodeID is derived from "pipeline type" and "component ID".
 type receiverNode struct {
```

Comment:
Should you remove the `component.Component` from the nodes?

Reply:
I think it should stay so that we can iterate over nodes and Start/Stop them without needing to resolve a more specific type.

Reply:
But is not initialized correct?

Reply:
Not yet but it should be very soon.

Reply:
#7026 can supersede this PR, but I'm curious whether you will allow temporarily untested code for the sake of having reasonably sized PRs.